app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt

```diff
@@ -0,0 +1,117 @@
+package uk.gov.android.authentication
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.ActivityCompat
+import java.util.UUID
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+
+@Suppress("TooGenericExceptionThrown")
+class AppAuthSession : LoginSession {
+    private var context: Context? = null
+    private lateinit var authService: AuthorizationService
+
+    override fun init(
+        context: Context
+    ): LoginSession {
+        if (this.context == null) {
+            this.context = context
+            authService = AuthorizationService(context)
+        }
+        return this
+    }
+
+    override fun present(
+        configuration: LoginSessionConfiguration
+    ) {
+        if (context == null) {
+            throw Error("Context is null, did you call init?")
+        }
+
+        with(configuration) {
+            val context = this@AppAuthSession.context!!
```

Comment:
Does the configuration have a context within it? If so, is there a reason we're choosing to create a variable for the explicit purpose of referring to the context within this AppAuthSession?

Reply:
Yeah there isn't, and putting the context into the constructor will mean it doesn't need to be nullable so also solves this. I expect it is maybe from hangover of original revisions of this function perhaps.... 🤷